src/inc/volatile.h

```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__ANDROID__))
```

Comment:
Would it be better to check against GCC version that added support for `__atomic_load`?